python/variable.cpp

```diff
@@ -193,7 +193,12 @@ void init_variable(py::module &m) {
   // This should be in the certain order
   bind_init_0D(variable);
   bind_init_0D<bool>(variable);
+#ifdef _WIN32
+  // C long 32 bit on windows.
+  bind_init_0D<int32_t>(variable);
```

Comment:
Are Python integers 32 bit on Windows?

Reply:
Yes https://docs.microsoft.com/en-us/cpp/c-language/storage-of-basic-types?redirectedfrom=MSDN&view=vs-2019 https://stackoverflow.com/questions/22513445/python-handles-long-ints-differently-on-windows-and-unix

Reply:
Are we sure that the `int64` constructor is not needed on Windows? This makes it impossible to store a 0D value larger than `2<<30 - 1`, so this `scipp.Variable(2<<30)` fails as it cannot find a representation. I doubt many things go that high, but maybe some log values could (e.g. timestamps).

Having both int32 and int64 seems to work OK:

```
In [2]: scipp.Variable(1)
Out[2]:     <scipp.Variable>          int32      [dimensionless]  ()  [1]

In [3]: scipp.Variable(2<<30)
Out[3]:     <scipp.Variable>          int64      [dimensionless]  ()  [2147483648]
```

It is however dependent on the order it is defined in the source. If the bind_init_0D<int64_t> is defined first then `scipp.Variable(1)` will bind to `int64`..

Reply:
@DTasev instead of the compile-time or overloaded route - maybe there is a smarter way to do this at runtime based on the size of the integer passed? I believe this is how python and numpy interoperate.

Reply:
I would object to runtime based size checks. Can't we just make it `int64_t` and do away with some of the special case handling? I would value *consistency within scipp* **higher** than *consistency with platform-specific Python/numpy behavior*.

Reply:
... in particular since no significant harm can come from using a *wider* type.

Reply:
`int64_t` can be made the default, the issue is - then every `np.arange` needs to have `dtype=np.int64` added, or tests that do `sc.Variable(N)` will fail their equality checks on Windows. 

(By the way `np.ones` gives `float64` by default on Windows, `numpy` is not fully consistent with itself either!)